ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,25 +1,24 @@
 """Serialization Strategy."""
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict
 
 from google.protobuf import type_pb2
-from google.protobuf.internal import decoder
-from google.protobuf.internal import encoder
+from google.protobuf.internal import decoder, encoder
+from typing_extensions import TypeAlias
 
+InnerEncoder: TypeAlias = Callable[[Callable[[bytes], int], bytes, bool], int]
+InnerDecoder: TypeAlias = Callable[
```

Comment:
[Typeshed](https://github.com/python/typeshed/blob/main/stubs/protobuf/google/protobuf/internal/decoder.pyi) has _Decoder as `_Decoder: TypeAlias = Callable[[str, int, int, Message, dict[FieldDescriptor, Any]], int]`. However, this flags our [one call of it](https://github.com/ni/measurementlink-python/blob/main/ni_measurementlink_service/_internal/parameter/serializer.py#L165) as having invalid arguments. We're passing 'memoryview' instead of 'str', 'ValueType' instead of 'Message', and 'dict[int, Any]' instead of 'dict[FieldDescriptor, Any]'.

Reply:
`memoryview` instead of `str`: I brought this up in the previous PR:

> I am not sure what type this should be (getbuffer() is a memory view, getvalue() is bytes) or whether the typeshed for google.protobuf.internal.decoder._Decoder is correct (it uses str?)

`ValueType` instead of `Message`: `Kind` is a protobuf enum. `Kind.ValueType` is an opaque alias for `int`: https://github.com/python/typeshed/blob/main/stubs/protobuf/google/protobuf/type_pb2.pyi#L121 . I don't understand how we are able to pass an int instead of a message. Either our code is incorrect or typeshed is incorrect.

`int` instead of `FieldDescriptor`: I brought this up in the previous PR:

> The 2nd parameter is key, which is typed as str in serialization_strategy.py.
>
>Here, we are passing an int.
>
>typeshed annotates key as a FieldDescriptor.
>
>Are these the same thing? Which is right?

By "are these the same thing" I meant "am I even looking at the correct type?"

Reply:
I looked into whether we can construct `FieldDescriptor`. We can't construct descriptors directly anymore, but if we use `descriptor.proto` types to add a `FileDescriptorProto` to the descriptor pool, then the protobuf package will generate classes at run time like it does when you import a `_pb2.py` file.

https://github.com/ni/measurementlink-python/issues/35